### PR TITLE
Use Gradle version catalog

### PR DESCRIPTION
**Pull request by snazy**
# Description

Introduces Gradle version catalogs (toml files), also bump a couple dependencies to recent versions.

Version catalogs as a central place to declare dependencies are easier to manage. Tools like renovate(bot) and dependabot can use these.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

File: extension/persistence/eclipselink/build.gradle [hunk 1]
```diff
@@ -17,9 +17,10 @@
 dependencies {
     implementation(project(":polaris-core"))
     implementation(project(":polaris-service"))
-    implementation("org.eclipse.persistence:eclipselink:4.0.3")
-    implementation("io.dropwizard:dropwizard-jackson:${dropwizardVersion}")
+    implementation(libs.eclipselink)
+    implementation(platform(libs.dropwizard.bom))
+    implementation("io.dropwizard:dropwizard-jackson")
```
Comment (RussellSpitzer) [inline extension/persistence/eclipselink/build.gradle:22]: Is there a reason this one doesn't have a "libs." identifier?

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+[libraries]
+assertj-core = { module = "org.assertj:assertj-core", version = "3.26.3" }
+auth0-jwt = { module = "com.auth0:java-jwt", version = "4.2.1" }
+awssdk-bom = { module = "software.amazon.awssdk:bom", version = "2.26.25" }
+azuresdk-bom = { module = "com.azure:azure-sdk-bom", version = "1.2.25" }
+bouncycastle-bcprov = { module = "org.bouncycastle:bcprov-jdk18on", version = "1.78" }
+caffeine = { module = "com.github.ben-manes.caffeine:caffeine", version = "3.1.8" }
+commons-codec1 = { module = "commons-codec:commons-codec", version = "1.17.0" }
+commons-lang3 = { module = "org.apache.commons:commons-lang3", version = "3.14.0" }
+dropwizard-bom = { module = "io.dropwizard:dropwizard-bom", version = "4.0.7" }
```
Comment (RussellSpitzer) [inline gradle/libs.versions.toml:34]: can't `version` here be `version.ref = "dropwizard"`

**Review by RussellSpitzer — COMMENTED** [review]
Minor nits, Looks good to me otherwise

File: extension/persistence/eclipselink/build.gradle (hunk 1 above)
Comment (snazy) [reply extension/persistence/eclipselink/build.gradle:22]: Yup - it's just some artifact that's "version managed" via a bom.

File: build.gradle [hunk 3]
```diff
@@ -72,17 +65,17 @@ subprojects {
...
-        implementation("org.slf4j:slf4j-api:2.0.12")
-        compileOnly("com.github.spotbugs:spotbugs-annotations:4.8.5")
+        implementation(libs.guava)
+        implementation(libs.slf4j.api)
+        compileOnly(libs.jetbrains.annotations)
+        compileOnly(libs.spotbugs.annotations)
 
-        testImplementation(platform("org.junit:junit-bom:5.10.3"))
+        testImplementation(platform(libs.junit.bom))
         testImplementation("org.junit.jupiter:junit-jupiter")
```
Comment (flyrain) [inline build.gradle:76]: Can we use `lib.xxx` here as well?

**Review by flyrain — APPROVED** [review]
LGTM, thanks @snazy !

File: build.gradle (hunk 3 above)
Comment (snazy) [reply build.gradle:76]: We could - it's a taste/preference thing.
There can be a lot of "unmanaged" dependencies (version-managed via a bom) mentioned in `libs.gradle.toml` over time.

Comment (snazy) [reply build.gradle:76]: WRT to "visible dependency report" - I'll come up w/ additional PRs to a) automatically licenses, b) verify `NOTICE` content and c) generate a license report containing all deps in distributables.

**Review by RussellSpitzer — APPROVED** [review]

File: build.gradle (hunk 3 above)
Comment (flyrain) [reply build.gradle:76]: > it's a taste/preference thing.

I'd prefer a consistent style, but we can fix it in followup PRs.

**Pull request merged**